lib/src/kbasesearchengine/events/handler/WorkspaceEventHandler.java

```diff
@@ -649,4 +678,143 @@ private static String toWSRefPath(final List<GUID> objectRefPath) {
         return String.join(";", refpath);
     }
 
+    @Override
+    public StatusEvent updateEvent(final StatusEvent ev)
+            throws IndexingException,
+                   RetriableIndexingException {
+
+        // only object level events are updated below
+        if (!ObjectEventQueue.OBJ_LVL_EVENTS.contains(ev.getEventType())) {
+            return ev;
+        }
+
+        // brute force get latest state and update event as event can be played
+        // out of order in the case of failed events being replayed.
+
+        // states to update
+        final StatusEventType latestEventType = ev.getEventType();
+        Boolean latestIsPublic = null;
+        String latestName = null;
+
+        if (ev.isPublic().isPresent()) {
+            latestIsPublic = ev.isPublic().get().booleanValue();
+        }
+        if (ev.getNewName().isPresent()) {
+            latestName = ev.getNewName().get().toString();
+        }
+
+
+        // wsid and objid of the object that the specified StatusEvent is an event for
+        final int wsid = ev.getAccessGroupId().get();
+        final long objid = Long.valueOf(ev.getAccessGroupObjectId().get().toString()).longValue();
+
+        Map<String, Object> command;
+
+        // check if workspace is permanently deleted or marked as deleted
+        try {
+            final Tuple9 tuple = getWorkspaceInfo(wsid);
+        } catch(IOException ex) {
```

Comment:
space after catch to differentiate keywords from methods

Reply:
Fixed.